src/integrity-checker.js

```diff
@@ -292,16 +300,19 @@ export default class InstallationIntegrityChecker {
         return 'FILES_MISSING';
       }
 
-      // Since we know the "files" entry is sorted alphabetically, we can optimize the thing
+      // Since we know the "files" entry is sorted (alphabetically), we can optimize the thing
       // Instead of storing the files in a Set, we can just iterate both arrays at once. O(n)!
       for (let u = 0, v = 0; u < expected.files.length; ++u) {
-        // Skip over files that have been added
-        while (v < actual.files.length && actual.files[v] !== expected.files[u]) {
+        // Number of iterations after which there won't be enough entries remaining for the arrays to match
+        const max = actual.files.length - expected.files.length;
+
+        // Skip over files that have been added (not present in 'expected')
+        while (v < max && actual.files[v] !== expected.files[u]) {
           v += 1;
         }
 
         // If we've reached the end of the actual array, the file is missing
-        if (v === actual.files.length) {
+        if (v === max) {
```

Comment:
Don't think this logic is correct. `max` is just a "small" difference where `v` is the absolute index which has to gi past `max`. If you kept a separate `advance` variable and calculated `v` as `u + advance` and did `if (advance === max)` then the logic would be correct.